Java/src/4-curry.java

```diff
@@ -0,0 +1,29 @@
+import java.util.function.Function;
+
+class Curry {
+    public static void main(String[] args) {
+        final TriFunction<Integer,Integer,Integer,Integer> add = Curry::sum;
+        Function<Integer, Function<Integer, Function<Integer,Integer>>> uncurry = Curry.curry(add);
+        TriFunction<Integer,Integer,Integer,Integer> curry = Curry.uncurry(uncurry);
+        //usage
+        System.out.println("Curried result : " + curry.apply(1,2,3));
+        System.out.println("Uncurried result : " + uncurry.apply(1).apply(2).apply(3));
+    }
+    private static Integer sum(Integer a, Integer b, Integer c){
+        return a + b + c;
+    }
+
+    private static <A, B, C, D> Function<A, Function<B, Function<C,D>>> curry(final TriFunction<A, B, C, D> f) {
+        return (A a) -> (B b) -> (C c) -> f.apply(a, b, c);
+    }
+
+    private static <A, B, C, D> TriFunction<A,B,C,D> uncurry(Function<A, Function<B, Function<C, D>>> f) {
```

Comment:
Could you split this line to fit 100 symbols in line, please? Just a code style suggestion.